todo/src/components/TodoList.js

```diff
@@ -0,0 +1,100 @@
+import React, {Component, Fragment} from 'react';
+import {connect} from 'react-redux';
+import { IoMdClose } from 'react-icons/io';
+
+import {markTaskComplete, clearCompleted, addTask, removeTask} from './actions'
+
+
+// Yes, yes, I know, this is one BIG component with ALL my redux crammed into it!
+// But, my argument here is, Redux is already a whole lot more typing. I can break these
+// into individual components, that's actually trivial at this point.
+// And I will definitely do that when we move into groups or on production sites.
+// But for this demo/learning tool, etc, it's MUCH simpler to just keep most functionality
+// in one file. Easier to reference, easier to see what's going on, and I think easier
```

Comment:
Easier until you have to come back to the code in 5 months and scan through a 200 line file and try to remember what everything is doing. :P When you break it down into smaller parts and give each part one particular job, it's easier to tell at a glance what that part's job is.